cmds/init/bgbuild.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2017 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Build commands in backgorund processes. This feature certainly makes a
+// non-busybox shell feel more realtime.
+package main
+
+import (
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"runtime"
+	"sort"
+)
+
+// Generally shorter commands (ls, cd, ...) are used more often, so they are
+// compiled first. This avoids maintaining a list for compilation order.
+type cmdSlice []os.FileInfo
+
+func (p cmdSlice) Len() int {
+	return len(p)
+}
+
+func (p cmdSlice) Less(i, j int) bool {
+	return len(p[i].Name()) < len(p[j].Name())
+}
+
+func (p cmdSlice) Swap(i, j int) {
+	p[i], p[j] = p[j], p[i]
+}
+
+func worker(cmds chan string) {
```

Comment:
One thought here. Is there any value in looking in the environment or something that would tell us whether we want to do this and, further, the priority? /proc/cmdline maybe?